test/bases/renku_data_services/data_api/test_projects.py

```diff
@@ -225,6 +225,42 @@ async def test_project_creation_with_conflicting_slug(sanic_client, user_headers
     assert response.status_code == 409, response.text
 
 
+@pytest.mark.asyncio
+async def test_project_creation_with_duplicate_repositories(sanic_client, user_headers, regular_user) -> None:
+    namespace = regular_user.namespace.slug
+    payload = {
+        "name": "My Project",
+        "namespace": namespace,
+        "repositories": [
+            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
+            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
```

Comment:
question: this would probably fail if it was
```suggestion
            "http://github.com/SwissDataScienceCenter/renku-data-services.git",
            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
```
but I'm not sure if we want to handle this case. In a way it's not a duplicate, but I also can't think of a reason why a user would want both http and https for the same repo, so preventing an edge case for them could be nice

Reply:
I think it's something that can be handled once we try to re-write repo URLs to use the canonical URL.